stdpopsim/cli.py

```diff
@@ -237,9 +237,12 @@ def write_bibtex(engine, model, contig, bibtex_file):
     Write bibtex for available citations to a file."""
     for citation in engine.citations:
         bibtex_file.write(citation.fetch_bibtex())
-
-    for citation in model.citations:
-        bibtex_file.write(citation.fetch_bibtex())
+    if contig.genetic_map is not None:
+        for citation in contig.genetic_map.citations:
+            bibtex_file.write(citation.fetch_bibtex())
+    if len(model.citations) > 0:
+        for citation in model.citations:
+            bibtex_file.write(citation.fetch_bibtex())
 
```

Comment:
I forgot to check this --- are we closing the bibtex file after we finish writing to it? We could just put a ``bibtex_file.close()`` at the end of the method.

Reply:
Right, sorry I had forgotten about this. I'll close it.